src/DocumentDB.ChangeFeedProcessor/PartitionManagement/PartitionSupervisor.cs

```diff
@@ -63,11 +63,16 @@ public async Task RunAsync(CancellationToken shutdownToken)
             {
                 closeReason = ChangeFeedObserverCloseReason.Shutdown;
             }
-            catch (Exception) when (processorTask.IsFaulted)
+            catch (Exception ex) when (processorTask.IsFaulted && ex is ObserverException)
```

Comment:
Hmm, looks like disappeared comment from last review. We shouldn't check for IsFaulted. Can we just make it catch(ObserverException)?

Reply:
Done